test/models/test_fully_bayesian.py

```diff
@@ -656,6 +666,77 @@ def test_custom_pyro_model(self):
                     atol=5e-4,
                 )
 
+    def test_condition_on_observation(self):
```

Comment:
Thanks for adding these tests. For completeness, can you add a test that calls `model.fantasize` and a test that evaluates some acquisition function (e.g., qLogEI) with the fantasized model? I just want to make sure everything works e2e including the sampler related code inside the acquisition functions.